pkg/trace/info/stats.go

```diff
@@ -155,15 +189,66 @@ func (ts *TagStats) publish() {
 	metrics.Count("datadog.trace_agent.receiver.events_extracted", eventsExtracted, tags, 1)
 	metrics.Count("datadog.trace_agent.receiver.events_sampled", eventsSampled, tags, 1)
 	metrics.Count("datadog.trace_agent.receiver.payload_accepted", requestsMade, tags, 1)
+
+
+	var droppedReasons []string
+	for _, c := range ts.droppedTraceReasonCounts() {
+		if c.count > 0 {
+			metrics.Count("datadog.trace_agent.normalizer.traces_dropped", c.count, append(tags, "reason:" + c.reason), 1)
+			droppedReasons = append(droppedReasons, c.reason + ":" + strconv.FormatInt(c.count, 10))
+		}
+	}
+
+	var malformedReasons []string
+	for _, c := range ts.malformedTraceReasonCounts() {
+		if c.count > 0 {
+			metrics.Count("datadog.trace_agent.normalizer.traces_malformed", c.count, append(tags, "reason:" + c.reason), 1)
+			malformedReasons = append(malformedReasons, c.reason + ":" + strconv.FormatInt(c.count, 10))
+		}
+	}
+
+	var normalizerMessages []string
+	if len(droppedReasons) > 0 {
+		normalizerMessages = append(normalizerMessages, fmt.Sprintf("dropped_traces(%s)", strings.Join(droppedReasons, ", ")))
+	}
+	if len(malformedReasons) > 0 {
+		normalizerMessages = append(normalizerMessages, fmt.Sprintf("malformed_traces(%s)", strings.Join(malformedReasons, ", ")))
+	}
+
+	if len(normalizerMessages) > 0 {
+		log.Warn("trace normalization problems detected: %s", strings.Join(normalizerMessages, " "))
+	}
 }
 
 // Stats holds the metrics that will be reported every 10s by the agent.
 // Its fields require to be accessed in an atomic way.
 type Stats struct {
 	// TracesReceived is the total number of traces received, including the dropped ones.
 	TracesReceived int64
+	// TracesDroppedDecodingError is the number of traces dropped due to decoding error
```

Comment:
This is really nasty. Would it be better if we used embedded structs? Perhaps:
```go
type TracesDropped struct {
    DecodingError int64
    EmptyTrace int64
    TraceIDZero
    // ...
}
```
It might be a bit better.

The same for `TracesMalformed`.

Reply:
Agreed, updated